portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,25 +33,34 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  private final List<String> entries = new ArrayList<String>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    String json = convertToJson(entries);
-    response.getWriter().println(json);
+    Query query = new Query("Entry");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Entry> entries = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String email = (String) entity.getProperty("email");
+      String comment = (String) entity.getProperty("comment");
+      Entry entry = new Entry(name, email, comment);
+      entries.add(entry);
+    }
+    System.out.println(entries);
```

Comment:
You should move everything from Query query till here to a separate method,

private List<Entry> entries = readFromDatastore();

Reply:
Done